server/datastore/mysql/apple_mdm.go

```diff
@@ -4789,3 +4790,351 @@ WHERE
+		return ctxerr.Wrap(ctx, err, "updating vpp token team")
+	}
+
+	return nil
+}
+
+func (ds *Datastore) DeleteVPPToken(ctx context.Context, tokenID uint) error {
+	_, err := ds.writer(ctx).ExecContext(ctx, `DELETE FROM vpp_tokens WHERE id = ?`, tokenID)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "deleting vpp token")
+	}
+
+	return nil
+}
+
+func (ds *Datastore) ListVPPTokens(ctx context.Context) ([]fleet.VPPTokenDB, error) {
+	stmt := `
+	SELECT
+		id,
+		organization_name,
+		location,
+		renew_at,
+		token,
+		team_id,
+		null_team_type
+	FROM
+		vpp_tokens
+`
+	var tokEncs []struct {
+		ID        uint               `db:"id"`
+		OrgName   string             `db:"organization_name"`
+		Location  string             `db:"location"`
+		RenewDate time.Time          `db:"renew_at"`
+		Token     []byte             `db:"token"`
+		TeamID    *uint              `db:"team_id"`
+		NullTeam  fleet.NullTeamType `db:"null_team_type"`
+	}
+
+	if err := sqlx.SelectContext(ctx, ds.reader(ctx), &tokEncs, stmt); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "selecting vpp tokens from db")
+	}
+
+	var tokens []fleet.VPPTokenDB
+
+	for _, tokEnc := range tokEncs {
+		tokDec, err := decrypt(tokEnc.Token, ds.serverPrivateKey)
+		if err != nil {
+			return nil, ctxerr.Wrap(ctx, err, "decrypting vpp token with serverPrivateKey")
+		}
+
+		tokens = append(tokens, fleet.VPPTokenDB{
+			ID:        tokEnc.ID,
+			OrgName:   tokEnc.OrgName,
+			Location:  tokEnc.Location,
+			RenewDate: tokEnc.RenewDate,
+			Token:     string(tokDec),
+			TeamID:    tokEnc.TeamID,
+			NullTeam:  tokEnc.NullTeam,
+		})
+	}
+
+	return tokens, nil
+}
+
+func (ds *Datastore) GetVPPTokenByTeamID(ctx context.Context, teamID *uint) (*fleet.VPPTokenDB, error) {
+	stmtTeam := `
+	SELECT
+		id,
+		organization_name,
+		location,
+		renew_at,
+		token,
+		team_id,
+		null_team_type
+	FROM
+		vpp_tokens
+	WHERE
+		team_id = ?
+`
+	stmtNullTeam := `
+	SELECT
+		id,
+		organization_name,
+		location,
+		renew_at,
+		token,
+		team_id,
+		null_team_type
+	FROM
+		vpp_tokens
+	WHERE
+		team_id IS NULL
+    AND
+        null_team_type = ?
+`
+
+	var tokEnc struct {
+		ID        uint               `db:"id"`
+		OrgName   string             `db:"organization_name"`
+		Location  string             `db:"location"`
+		RenewDate time.Time          `db:"renew_at"`
+		Token     []byte             `db:"token"`
+		TeamID    *uint              `db:"team_id"`
+		NullTeam  fleet.NullTeamType `db:"null_team_type"`
+	}
+
+	var err error
+	if teamID != nil {
+		err = sqlx.GetContext(ctx, ds.reader(ctx), &tokEnc, stmtTeam, teamID)
+	} else {
+		err = sqlx.GetContext(ctx, ds.reader(ctx), &tokEnc, stmtNullTeam, fleet.NullTeamNoTeam)
+	}
+	if err != nil {
+		if errors.Is(sql.ErrNoRows, err) {
+			if err := sqlx.GetContext(ctx, ds.reader(ctx), &tokEnc, stmtNullTeam, fleet.NullTeamAllTeams); err != nil {
+				return nil, ctxerr.Wrap(ctx, err, "retrieving vpp token by team")
```

Comment:
Usually in the case of `sql.ErrNoRows`, we wrap the returned error in a `notFound` so it gets properly detected by the Service layer and results in `Not Found` status codes in HTTP. Depends on how/where this gets called, but it should probably have this anyway (you can search the Datastore layer for `if err == sql.ErrNoRows` for examples).